crates/ruff_linter/src/rules/pyupgrade/rules/pep695/private_type_parameter.rs

```diff
@@ -133,7 +134,34 @@ pub(crate) fn private_type_parameter(checker: &Checker, binding: &Binding) -> Op
         return Some(diagnostic);
     }
 
-    diagnostic.try_set_fix(|| {
+    diagnostic.try_set_optional_fix(|| {
+        let source = checker.source();
+
+        // Copied from PYI019
+        let reference_is_complex = |id: ResolvedReferenceId| -> bool {
+            let reference = semantic.reference(id);
+            let in_source = &source[reference.range()];
+
+            in_source != old_name
+        };
+
+        if !is_identifier(new_name) {
+            return Ok(None);
+        }
+
+        let existing_binding_for_new_name = semantic.simulate_runtime_load_at_location_in_scope(
+            new_name,
+            binding.range,
+            binding.scope,
+            TypingOnlyBindingsStatus::Disallowed,
+        );
+
+        if let Some(binding) = existing_binding_for_new_name.map(|id| semantic.binding(id)) {
+            if binding.kind.is_type_param() {
+                return Ok(None);
+            }
+        }
+
```

Comment:
I don't think checking this again is necessary; no tests fail if I remove this added code. We already check this on line 133 with the `ShadowedKind` logic.